### PR TITLE
CLOUDP-171400 Add crud commands "atlas dataLakePipelines"

### PR DESCRIPTION
**Pull request by fmenezes**
## Proposed changes

Add crud commands "atlas dataLakePipelines" 

_Jira ticket:_ CLOUDP-171400

## Checklist

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -112,6 +112,7 @@ Review and replace command name and arguments depending on the command you are u
             "mode": "auto",
             "program": "${workspaceFolder}/cmd/atlas",
             "env": {},
+            "buildFlags": "-ldflags '-X github.com/mongodb/mongodb-atlas-cli/internal/config.ToolName=atlascli'",
```
Comment (fmenezes) [inline CONTRIBUTING.md:115]: flyby

File: internal/cli/atlas/data_lake/pipelines/create_test.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	"github.com/mongodb/mongodb-atlas-cli/internal/mocks"
+	"github.com/mongodb/mongodb-atlas-cli/internal/test"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCreate_Run(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockStore := mocks.NewMockPipelinesCreator(ctrl)
+
+	var expected interface{} // TODO change here
```
Comment (colm-quinn) [inline internal/cli/atlas/data_lake/pipelines/create_test.go:35]: 👀 Does expected need an update?

File: internal/cli/atlas/data_lake/pipelines/update_test.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	"github.com/mongodb/mongodb-atlas-cli/internal/mocks"
+	"github.com/mongodb/mongodb-atlas-cli/internal/test"
+	"github.com/stretchr/testify/require"
+)
+
+func TestUpdate_Run(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockStore := mocks.NewMockPipelinesUpdater(ctrl)
+
+	var expected interface{} // TODO change here
```
Comment (colm-quinn) [inline internal/cli/atlas/data_lake/pipelines/update_test.go:35]: 👀

File: internal/cli/atlas/data_lake/pipelines/describe.go [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+	return opts.Print(r)
+}
+
+// atlas dataLake pipelines describe <pipelineName> [--projectId projectId].
+func DescribeBuilder() *cobra.Command {
+	opts := new(DescribeOpts)
+	cmd := &cobra.Command{
+		Use:   "describe <pipelineName>",
+		Short: "Describes the Data Lake Pipeline. To use this resource, the requesting API Key must have the Project Read Only role..",
```
Comment (colm-quinn) [inline internal/cli/atlas/data_lake/pipelines/describe.go:65]:
```suggestion
		Short: "Describes the Data Lake Pipeline. To use this resource, the requesting API Key must have the Project Read Only role.",
```

File: internal/cli/atlas/data_lake/pipelines/create.go [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+		if len(entries) != 2 || len(entries[0]) == 0 || len(entries[1]) == 0 {
+			return fmt.Errorf("%w: expected format is 'type:field' got '%s'", ErrSinkTransformInvalid, entry)
+		}
+	}
+
+	return nil
+}
+
+func (opts *CreateOpts) newCreateRequest() (*atlasv2.IngestionPipeline, error) {
+	if len(opts.filename) > 0 {
```
Comment (gssbzn) [inline internal/cli/atlas/data_lake/pipelines/create.go:113]:
```suggestion
	if opts.filename != "" {
```
should be equivalent and simpler

File: internal/cli/atlas/data_lake/pipelines/create.go [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+		pipeline := &atlasv2.IngestionPipeline{}
+		if err := file.Load(opts.fs, opts.filename, pipeline); err != nil {
+			return nil, err
+		}
+		return pipeline, nil
```
Comment (gssbzn) [inline internal/cli/atlas/data_lake/pipelines/create.go:118]: seems like file and other flags are mutually exclusive, can we mark the flags as such?

File: internal/usage/usage.go [hunk 7]
```diff
@@ -404,4 +404,14 @@ dbName and collection are required only for built-in roles.`
...
+	SinkType                                  = "Type of ingestion destination of this Data Lake Pipeline."
+	SinkMetadataProvider                      = "Target cloud provider for this Data Lake Pipeline."
+	SinkMetadataRegion                        = "Target cloud provider region for this Data Lake Pipeline."
+	SinkPartitionField                        = "Ordered fields used to physically organize data in the destination."
+	SourceType                                = "Type of ingestion source of this Data Lake Pipeline."
+	SourceClusterName                         = "Human-readable name that identifies the cluster."
+	SourceCollectionName                      = "Human-readable name that identifies the collection."
+	SourceDatabaseName                        = "Human-readable name that identifies the database."
+	Transform                                 = "Fields to be excluded for this Data Lake Pipeline."
+	PipelineFilename                          = "Name of the JSON index configuration file to use."
```
Comment (gssbzn) [inline internal/usage/usage.go:416]: 👀 copy/paste ?

File: internal/cli/atlas/data_lake/pipelines/pipelines.go [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+package pipelines
+
+import (
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli"
+	"github.com/spf13/cobra"
+)
+
+const periodicCPS = "PERIODIC_CPS"
+const onDemandCPS = "ON_DEMAND_CPS"
```
Comment (gssbzn) [inline internal/cli/atlas/data_lake/pipelines/pipelines.go:25]: can you offer auto-complete for these?

File: internal/usage/usage.go (hunk 7 above)
Comment (fmenezes) [reply internal/usage/usage.go:416]: yes

File: internal/cli/atlas/data_lake/pipelines/create.go (hunk 6 above)
Comment (fmenezes) [reply internal/cli/atlas/data_lake/pipelines/create.go:118]: I did, on line 210

File: internal/cli/atlas/data_lake/pipelines/describe.go [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+// atlas dataLake pipelines describe <pipelineName> [--projectId projectId].
+func DescribeBuilder() *cobra.Command {
+	opts := new(DescribeOpts)
+	cmd := &cobra.Command{
+		Use:   "describe <pipelineName>",
+		Short: "Describes the Data Lake Pipeline. To use this resource, the requesting API Key must have the Project Read Only role.",
+		Long:  fmt.Sprintf(usage.RequiredRole, "Project Owner"),
+		Args:  require.ExactArgs(1),
+		Annotations: map[string]string{
+			"pipelineNameDesc": "Name of the pipeline",
```
Comment (sarahsimpers) [inline internal/cli/atlas/data_lake/pipelines/describe.go:69]:
```suggestion
			"pipelineNameDesc": "Label that identifies the pipeline",
```

File: internal/cli/atlas/data_lake/pipelines/pipelines.go [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+func autoCompleteSourceType(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	return []string{periodicCPS, onDemandCPS}, cobra.ShellCompDirectiveDefault
+}
+
+func Builder() *cobra.Command {
+	const use = "pipelines"
+	cmd := &cobra.Command{
+		Use:     use,
+		Aliases: cli.GenerateAliases(use),
+		Short:   "Data Lake Pipelines.",
```
Comment (sarahsimpers) [inline internal/cli/atlas/data_lake/pipelines/pipelines.go:36]:
```suggestion
		Short:   "Data lake pipelines.",
```

File: internal/usage/usage.go [hunk 11]
```diff
@@ -404,4 +404,14 @@ dbName and collection are required only for built-in roles.`
...
+	SinkType                                  = "Type of ingestion destination of this Data Lake Pipeline."
+	SinkMetadataProvider                      = "Target cloud provider for this Data Lake Pipeline."
+	SinkMetadataRegion                        = "Target cloud provider region for this Data Lake Pipeline."
+	SinkPartitionField                        = "Ordered fields used to physically organize data in the destination."
+	SourceType                                = "Type of ingestion source of this Data Lake Pipeline."
+	SourceClusterName                         = "Human-readable name that identifies the cluster."
+	SourceCollectionName                      = "Human-readable name that identifies the collection."
+	SourceDatabaseName                        = "Human-readable name that identifies the database."
+	Transform                                 = "Fields to be excluded for this Data Lake Pipeline."
+	PipelineFilename                          = "Name of the JSON data lake pipeline configuration file to use."
```
Comment (sarahsimpers) [inline internal/usage/usage.go:416]:
```suggestion
	SinkType                                  = "Type of ingestion destination for this data lake pipeline."
	SinkMetadataProvider                      = "Target cloud provider for this data lake pipeline."
	SinkMetadataRegion                        = "Target cloud provider region for this data lake pipeline."
	SinkPartitionField                        = "Ordered fields used to physically organize data in the destination."
	SourceType                                = "Type of ingestion source for this data lake pipeline."
	SourceClusterName                         = "Human-readable label that identifies the source cluster."
	SourceCollectionName                      = "Human-readable label that identifies the source collection."
	SourceDatabaseName                        = "Human-readable label that identifies the source database."
	Transform                                 = "Fields to be excluded for this data lake pipeline."
	PipelineFilename                          = "Name of the JSON data lake pipeline configuration file to use."
```

File: internal/cli/atlas/data_lake/pipelines/list.go [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+// atlas dataLake pipelines list [--projectId projectId].
+func ListBuilder() *cobra.Command {
+	opts := &ListOpts{}
+	cmd := &cobra.Command{
+		Use:     "list",
+		Short:   "Returns a list of Data Lake Pipelines. To use this resource, the requesting API Key must have the Project Read Only role.",
+		Long:    fmt.Sprintf(usage.RequiredRole, "Project Read Only"),
+		Aliases: []string{"ls"},
+		Args:    require.NoArgs,
+		Example: ``,
```
Comment (sarahsimpers) [inline internal/cli/atlas/data_lake/pipelines/list.go:69]: Can we add an example?

**Review by sarahsimpers — CHANGES_REQUESTED** [review]
Some changes requested for docs, thank you!

**Review by gssbzn — APPROVED** [review]
LGTM

**Review by sarahsimpers — APPROVED** [review]
LGTM, thanks for all the changes!

**Comment by fmenezes** [comment]
for some reason docs test is stuck I'll try rebasing it

**Review by gssbzn — APPROVED** [review]
LGTM

**Pull request merged**